### PR TITLE
refactor(dotenv): Move some JS parse logic to RegExp. 

### PR DESCRIPTION
**Pull request by sevenwithawp**
Add support for Multiline and export variables

Comments:
Hi Deno Team! Excellent job on the project. I'm new to open-source so any comments what is not right will be welcome. After this merge I can start: dotenv-expand #2274

### CODE REVIEW

**Comment by sevenwithawp** [comment]
Pushed the second commit which is implementing dotenv-expand functionality #2274. Please let me know if something needs to be changed

**Review by kt3k — COMMENTED** [review]
Expanding feature looks nicely implemented and tested! Nice work!

Can you also add docs for expansion rules in README? `dotenv-expand` seems explaining the rule very well https://github.com/motdotla/dotenv-expand#what-rules-does-the-expansion-engine-follow

Also left a few questions

**Comment by sevenwithawp** [comment]
@kt3k Sure, I will add new rules to the readme. Also, you've made a valid point about the regexp. I will double-check and will add a few tests for these cases too. It will be done in upcoming days. Thanks!

File: dotenv/mod.ts [hunk 1]
```diff
@@ -20,30 +20,46 @@ export interface ConfigOptions {
...
+  key: string;
+  unquoted: string;
+  interpolated: string;
+  notInterpolated: string;
+};
+
+type CharactersMap = { [key: string]: string };
+
+const RE_KeyValue =
+  /^\s*(?:export\s+)?(?<key>[a-zA-Z_]+[a-zA-Z0-9_]*?)\s*=[\ \t]*('(?:\n)?(?<notInterpolated>(.|\n)*?)(?:\n)?'|"(?:\n)?(?<interpolated>(.|\n)*?)(?:\n)?"|(?<unquoted>[^\n#]*))\ *#*[\ \w]*$/gm;
```
[the remark this answers is not in the document]
Comment (sevenwithawp) [reply dotenv/mod.ts]: You were right about both cases

**Comment by sevenwithawp** [comment]
Hi @kt3k, There is an error in 1 from 12 PR checks. The error is:
`serveDirIndex TLS => ./http/file_server_test.ts:420:6 
error: ConnectionRefused: Connection refused (os error 61)
    const conn = await Deno.connectTls({`
Is something required from my side to solve that?

**Comment by bartlomieju** [comment]
> Hi @kt3k, There is an error in 1 from 12 PR checks. The error is: `serveDirIndex TLS => ./http/file_server_test.ts:420:6 error: ConnectionRefused: Connection refused (os error 61) const conn = await Deno.connectTls({` Is something required from my side to solve that?

Looks like a flake, I restarted the CI for you

**Comment by sevenwithawp** [comment]
> > Hi @kt3k, There is an error in 1 from 12 PR checks. The error is: `serveDirIndex TLS => ./http/file_server_test.ts:420:6 error: ConnectionRefused: Connection refused (os error 61) const conn = await Deno.connectTls({` Is something required from my side to solve that?
> 
> Looks like a flake, I restarted the CI for you

Thanks!

**Review by kt3k — APPROVED** [review]
LGTM. Great enhancement! Thanks for your contribution! @sevenwithawp

**Pull request merged**